### PR TITLE
SorrTask394_Convert_Buildmeister_process_gDoc_to_Markdown

### PR DESCRIPTION
**Pull request by zli00185**
#394

### CODE REVIEW

File: docs/Buildmeister_process.md [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+- [Post-mortem analysis (TBD)](#post-mortem-analysis-tbd)
+
+<!-- tocstop -->
+
+# General
+
+- Buildmeister rotates every 2 weeks
+  - To see who is the Buildmeister now refer to [<span
+    class="underline">Buildmeister gsheet</span>](https://docs.google.com/spreadsheets/d/1Ab6a3BVeLX1l1B3_A6rNY9pHRsofeoCw2ip2dkQ6SdA/edit#gid=0)
+
```
Comment (samarth9008) [inline docs/Buildmeister_process.md:18]: Remove this extra space.

File: docs/Buildmeister_process.md [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+      - [<span class="underline">Fast tests</span>](https://github.com/cryptokaizen/cmamp/actions/workflows/fast_tests.yml)
+      - [<span class="underline">Slow tests</span>](https://github.com/cryptokaizen/cmamp/actions/workflows/slow_tests.yml)
+    - dev_tools:
+      - [<span class="underline">Fast tests</span>](https://github.com/alphamatic/dev_tools/actions/workflows/fast_tests.yml)
+      - [<span class="underline">Slow tests</span>](https://github.com/alphamatic/dev_tools/actions/workflows/slow_tests.yml)
+  - Refer to .github dir in the repo for update schedule of GH actions
+  - Additional information about the [<span
+    class="underline">tests</span>](https://github.com/alphamatic/amp/blob/master/documentation/general/unit_tests.md#running-unit-tests)
+    and [<span
+    class="underline">gdoc</span>](https://docs.google.com/document/d/1M8I2qt5CuCw7537_1yayeZhIfh_VoeHT26h5jgSEm3Q/edit)
```
Comment (samarth9008) [inline docs/Buildmeister_process.md:40]: Update this link to `https://github.com/sorrentum/sorrentum/blob/master/docs/Unit_tests.md `

File: docs/Buildmeister_process.md [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+
+# Notification system
+
+- GH_bot notifies the team about breaks via Telegram channel @ALL
+
```
Comment (samarth9008) [inline docs/Buildmeister_process.md:45]: Remove extra space.

File: docs/Buildmeister_process.md [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+- A notification contains:
+  - Failing tests type: fast/slow/super-slow
+  - Repo
+  - Branch
+  - Event
+  - Link to a failing run
+
+Example:
+
+- <img src="Buildmeister_process_figs/image1.png" style="width:3.40625in;height:1.82292in" />
```
Comment (samarth9008) [inline docs/Buildmeister_process.md:55]: Image is not displayed.

File: docs/Buildmeister_process.md [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+
+# Buildmeister instructions
+
+- You receive a break notification from @GH_bot
+
```
Comment (samarth9008) [inline docs/Buildmeister_process.md:60]: Remove space. Remove everywhere from here. I will not point it out.

File: docs/Buildmeister_process.md [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+- Have a look at the message
+  - Do it right away, this is always your highest priority task
+
+- Notify the team
+
+Post on the @ALL Telegram channel what tests broke, e.g.,
```
Comment (samarth9008) [inline docs/Buildmeister_process.md:66]: Add `-` so that it starts with bullet point.

File: docs/Buildmeister_process.md [hunk 7]
```diff
@@ -0,0 +1,206 @@
...
+  - Do a quick run locally for failed test
+  - If test is specific and can not be run locally, rerun the regressions
+
+- Ask if somebody knows what is the problem
+  - If you know who is in charge of that test (you can use git blame) ask
+    directly
+
+- If the offender says that it's fixing the bug right away, let him / her do it
+
+- Otherwise file a bug to track the issue
```
Comment (samarth9008) [inline docs/Buildmeister_process.md:79]: They are are sub-points of above section.

File: docs/Buildmeister_process.md [hunk 8]
```diff
@@ -0,0 +1,206 @@
...
+      class="underline">https://github.com/alphamatic/dev_tools/actions/runs/1497955663</span>](https://github.com/alphamatic/dev_tools/actions/runs/1497955663)
+  - Provide as much information as possible to give an understanding of the
+    problem
+  - List all the tests with FAILED status in a github run, e.g.,  
+    FAILED knowledge_graph/vendors/test/test_p1_utils.py::TestClean::test_clean
+
+    FAILED
+    knowledge_graph/vendors/nbsc/test/test_nbsc_utils.py::TestExposeNBSCMetadata::test_expose_nbsc_metadata
+
+Stack trace or part of it (if it's too large)
```
Comment (samarth9008) [inline docs/Buildmeister_process.md:98]: Lets make this a sub-point of previous section.

File: docs/Buildmeister_process.md [hunk 9]
```diff
@@ -0,0 +1,206 @@
...
+"/.../knowledge_graph/vendors/utils.py", line 53, in load_release % version,
+File "/.../amp/helpers/dbg.py", line 335, in dassert_dir_exists \_dfatal(txt,
+msg, \*args) File "/.../amp/helpers/dbg.py", line 97, in \_dfatal
+dfatal(dfatal_txt) File "/.../amp/helpers/dbg.py", line 48, in dfatal raise
+assertion_type(ret) AssertionError:
+##############################################################################
+
+- Failed assertion \* dir='/fsx/research/data/kg/releases/timeseries_db/v0.5.2'
+  doesn't exist or it's not a dir The requested version 0.5.2 has no directory
+  associated with it.
```
Comment (samarth9008) [inline docs/Buildmeister_process.md:111]: Inside code format.

File: docs/Buildmeister_process.md [hunk 10]
```diff
@@ -0,0 +1,206 @@
...
+
+* Add the issue to the
+  [<span class="underline">BUILD - Breaks</span>](https://app.zenhub.com/workspaces/cm-615371012ed326001e044788/issues/cryptokaizen/cmamp/167)
+  Epic so that we can track it
+
+* If the failures are not connected to each other, file separate issues for each
+  of the potential root cause
+
+* Keep issues grouped according to the codebase organization
```
Comment (samarth9008) [inline docs/Buildmeister_process.md:120]: Sub points.

File: docs/Buildmeister_process.md [hunk 11]
```diff
@@ -0,0 +1,206 @@
...
+### `update_amp_submodule` fails
+
+- When this happens, the first thing to do is attempt to update the `amp`
+  pointer manually
+
+- Instructions:
+
+> cd src/dev_tools1 git checkout master git pull --recurse-submodules cd amp git
+> checkout master git pull origin master cd .. git add "amp" git commit -m
+> "Update amp pointer"
```
Comment (samarth9008) [inline docs/Buildmeister_process.md:159]: Code format.

**Review by samarth9008 — COMMENTED** [review]
One common improvement which I have not pointed out is the text which have different font in the word doc will go inside the code format. E.g. 
```
FAILED knowledge_graph/vendors/test/test_p1_utils.py::TestClean::test_clean
FAILED knowledge_graph/vendors/nbsc/test/test_nbsc_utils.py::TestExposeNBSCMetadata::test_expose_nbsc_metadata
```

File: docs/Buildmeister_process.md [hunk 12]
```diff
@@ -0,0 +1,191 @@
...
+
+<!-- tocstop -->
+
+# General
+
+- Buildmeister rotates every 2 weeks
+  - To see who is the Buildmeister now refer to [<span class="underline">Buildmeister gsheet</span>](https://docs.google.com/spreadsheets/d/1Ab6a3BVeLX1l1B3_A6rNY9pHRsofeoCw2ip2dkQ6SdA/edit#gid=0)
+  - Each rotation should be confirmed by a 'handshake' between the outgoing
+    Buildmeister and the new one in the related Telegram chat
+
```
Comment (samarth9008) [inline docs/Buildmeister_process.md:19]: Remove this space.

File: docs/Buildmeister_process.md [hunk 13]
```diff
@@ -0,0 +1,191 @@
...
+- Buildmeister is responsible for:
+  - Pushing team members to fix broken tests
+  - Conducting post-mortem analysis
+    - Why did the break happen?
+    - What can we avoid the problem, especially through process and automation
+
```
Comment (samarth9008) [inline docs/Buildmeister_process.md:25]: Remove this space.

File: docs/Buildmeister_process.md [hunk 14]
```diff
@@ -0,0 +1,191 @@
...
+- Testing workflows are available via github actions:
+  - Testing workflows that need to be checked:
+    - cmamp:
+      - [<span class="underline">Fast tests</span>](https://github.com/cryptokaizen/cmamp/actions/workflows/fast_tests.yml)
+      - [<span class="underline">Slow tests</span>](https://github.com/cryptokaizen/cmamp/actions/workflows/slow_tests.yml)
+    - dev_tools:
+      - [<span class="underline">Fast tests</span>](https://github.com/alphamatic/dev_tools/actions/workflows/fast_tests.yml)
+      - [<span class="underline">Slow tests</span>](https://github.com/alphamatic/dev_tools/actions/workflows/slow_tests.yml)
+  - Refer to .github dir in the repo for update schedule of GH actions
+  - Additional information about the [<spanclass="underline">tests</span>](https://github.com/sorrentum/sorrentum/blob/master/docs/Unit_tests.md) and [<spanclass="underline">gdoc</span>](https://docs.google.com/document/d/1M8I2qt5CuCw7537_1yayeZhIfh_VoeHT26h5jgSEm3Q/edit)
```
Comment (samarth9008) [inline docs/Buildmeister_process.md:35]: This is not properly formatted.

File: docs/Buildmeister_process.md [hunk 15]
```diff
@@ -0,0 +1,191 @@
...
+  - Paste the URL of the failing run
+    - Example: [<span class="underline">https://github.com/alphamatic/dev_tools/actions/runs/1497955663</span>](https://github.com/alphamatic/dev_tools/actions/runs/1497955663)
+  - Provide as much information as possible to give an understanding of the
+    problem
+  - List all the tests with FAILED status in a github run, e.g.,  
+    ```
+    FAILED knowledge_graph/vendors/test/test_p1_utils.py::TestClean::test_clean
+    
+    FAILED
+    knowledge_graph/vendors/nbsc/test/test_nbsc_utils.py::TestExposeNBSCMetadata::test_expose_nbsc_metadata
```
Comment (samarth9008) [inline docs/Buildmeister_process.md:82]: Keep this on a single line.

File: docs/Buildmeister_process.md [hunk 16]
```diff
@@ -0,0 +1,191 @@
...
+        "/.../knowledge_graph/vendors/utils.py", line 53, in load_release % version,
+        File "/.../amp/helpers/dbg.py", line 335, in dassert_dir_exists \_dfatal(txt,
+        msg, \*args) File "/.../amp/helpers/dbg.py", line 97, in \_dfatal
+        dfatal(dfatal_txt) File "/.../amp/helpers/dbg.py", line 48, in dfatal raise
+        assertion_type(ret) AssertionError:
+        ##############################################################################
+        * Failed assertion * dir='/fsx/research/data/kg/releases/timeseries_db/v0.5.2'
+        doesn't exist or it's not a dir The requested version 0.5.2 has no directory
+        associated with it.
+        ```
```
Comment (samarth9008) [inline docs/Buildmeister_process.md:98]: It's a new point and not a sub-point.

File: docs/Buildmeister_process.md (hunk 7 above)
Comment (zli00185) [reply docs/Buildmeister_process.md]: In the Gdoc "FALIED" is separated in a single line. Do I still need to follow the format there?

Comment (samarth9008) [reply docs/Buildmeister_process.md]: I think you commented on the wrong conversation but yes keep that on the same line.

**Comment by samarth9008** [comment]
Please resolve all the conversation and then only request for review. There are still changes which are not implemented.

**Comment by zli00185** [comment]
> I made it the sub-point for the List all the tests with FAILED status in a github run, e.g above. What is the format for the sub-point you refers to?

Could you response to this follow up question please?

**Comment by samarth9008** [comment]
Can you please point me to the comment you are mentioning?

**Comment by zli00185** [comment]
> Can you please point me to the comment you are mentioning?

I posted the comment here. Could you see it?
Screenshot 2023-07-11 at 1 17 28 AM

**Comment by samarth9008** [comment]
No need to make it a sub-point. `Stack trace or part of it (if it's too large)` starts with a new bullet point. Refer to the doc for proper formatting.

**Review by samarth9008 — APPROVED** [review]
LGTM

**Comment by samarth9008** [comment]
@DanilYachmenev PTAL

File: docs/Buildmeister_process.md [hunk 17]
```diff
@@ -0,0 +1,187 @@
...
+      - [<span class="underline">Slow tests</span>](https://github.com/cryptokaizen/cmamp/actions/workflows/slow_tests.yml)
+    - dev_tools:
+      - [<span class="underline">Fast tests</span>](https://github.com/alphamatic/dev_tools/actions/workflows/fast_tests.yml)
+      - [<span class="underline">Slow tests</span>](https://github.com/alphamatic/dev_tools/actions/workflows/slow_tests.yml)
+  - Refer to .github dir in the repo for update schedule of GH actions
+  - Additional information about the [<span class="underline">tests</span>](https://github.com/sorrentum/sorrentum/blob/master/docs/Unit_tests.md) and [<span class="underline">gdoc</span>](https://docs.google.com/document/d/1M8I2qt5CuCw7537_1yayeZhIfh_VoeHT26h5jgSEm3Q/edit)
+
+# Notification system
+
+- `GH_bot` notifies the team about breaks via Telegram channel `@ALL`
```
Comment (DanilYachmenev) [inline docs/Buildmeister_process.md:37]: `@ALL` -> `CK build notifications`

here and for other instances

Comment (DanilYachmenev) [inline docs/Buildmeister_process.md:37]: `GH_bot` -> `@CK_cmamp_buildbot`

here and for other instances

File: docs/Buildmeister_process.md [hunk 18]
```diff
@@ -0,0 +1,187 @@
...
+    - If test is specific and can not be run locally, rerun the regressions
+  - Ask if somebody knows what is the problem
+    - If you know who is in charge of that test (you can use `git blame`) ask
+        directly
+  - If the offender says that it's fixing the bug right away, let him / her do it
+  - Otherwise file a bug to track the issue
+
+- File an Issue in GH / ZH to report the failing tests and the errors
+  - Example:  
+    [<span class="underline">https://app.zenhub.com/workspaces/cm-615371012ed326001e044788/issues/alphamatic/dev_tools/318</span>](https://app.zenhub.com/workspaces/cm-615371012ed326001e044788/issues/alphamatic/dev_tools/318)
```
Comment (DanilYachmenev) [inline docs/Buildmeister_process.md:68]: lets replace example with https://github.com/cryptokaizen/cmamp/issues/4386
Just to keep it more up-to-date

File: docs/Buildmeister_process.md [hunk 19]
```diff
@@ -0,0 +1,187 @@
...
+    > git checkout master 
+    > git pull origin master 
+    > cd .. 
+    > git add "amp" 
+    > git commit -m "Update amp pointer"
+    ```
+There is also an invoke target `git_roll_amp_forward` that does an equivalent
+operation
+
+# Post-mortem analysis (TBD)
```
Comment (DanilYachmenev) [inline docs/Buildmeister_process.md:145]: @gpsaggese do we want to keep it?
I personally forgot the last time we updated it so maybe we should bring back the practise

**Review by DanilYachmenev — COMMENTED** [review]
gj!

lets update some info with more actual data but apart from that all looks good

**Comment by samarth9008** [comment]
@zli00185 It will be better for you to get done with these task first.

**Comment by DanilYachmenev** [comment]
@zli00185 could you pls resolve the conflict and ping me

**Comment by zli00185** [comment]
@DanilYachmenev Done. READ_ME file deleted due to the update of master branch.

**Comment by samarth9008** [comment]
So we have moved the Readme file to the root of the repo so mention the reference of this doc over there as you did earlier.

**Review by samarth9008 — APPROVED** [review]
LG

File: docs/Buildmeister_process.md (hunk 19 above)
Comment (gpsaggese) [reply docs/Buildmeister_process.md:145]: Let's leave it for now, but yeah it's probably not needed (this makes sense when the breaks are rare)

**Comment by DanilYachmenev** [comment]
@zli00185 another merge conflict here
pls fix and I'll merge

**Comment by zli00185** [comment]
> @zli00185 another merge conflict here
> pls fix and I'll merge

fixed

**Review by samarth9008 — APPROVED** [review]
LG

**Pull request merged**